programs/drift/src/instructions/keeper.rs

```diff
@@ -557,14 +570,54 @@ pub fn place_swift_taker_order<'c: 'info, 'info>(
             ..PlaceOrderOptions::default()
         },
     )?;
-    let non_matching_orders: Vec<OrderParams> = taker_order_params[1..].to_vec();
-    for non_matching_order_param in non_matching_orders {
-        if non_matching_order_param.order_type != OrderType::TriggerLimit
-            && non_matching_order_param.order_type != OrderType::TriggerMarket
-        {
-            msg!("Non-matching orders must be trigger orders");
-            return Err(print_error!(ErrorCode::SwiftOrderSequenceError)().into());
-        }
+
+    if let Some(stop_loss_order_params) = taker_order_params_message.stop_loss_order_params {
+        let stop_loss_order = OrderParams {
+            order_type: OrderType::TriggerMarket,
+            direction: matching_taker_order_params.direction.opposite(),
+            trigger_price: Some(stop_loss_order_params.trigger_price),
+            base_asset_amount: stop_loss_order_params.base_asset_amount,
+            trigger_condition: if matching_taker_order_params.direction == PositionDirection::Long {
+                OrderTriggerCondition::Below
+            } else {
+                OrderTriggerCondition::Above
+            },
+            market_index,
+            market_type: MarketType::Perp,
```

Comment:
we should throw error somewhere if market type from main order isnt perp?

Reply:
yeah